contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,661 @@
+        _;
+    }
+
+    modifier onlyAdminACL(bytes4 _selector) {
+        require(
+            adminACLAllowed(msg.sender, address(this), _selector),
+            "Only Admin ACL allowed"
+        );
+        _;
+    }
+
+    modifier onlyExistingDependencyType(bytes32 _dependencyTypeId) {
+        require(
+            _dependencyTypes.contains(_dependencyTypeId),
+            "Dependency type does not exist"
+        );
+        _;
+    }
+
+    /**
+     * @notice Initializes contract.
+     * @param _adminACLContract Address of admin access control contract, to be
+     * set as contract owner.
+     */
+    constructor(address _adminACLContract) {
+        // set AdminACL management contract as owner
+        _transferOwnership(_adminACLContract);
+    }
+
+    /**
+     * @notice Adds a new dependency type.
+     * @param _dependencyTypeId Name of dependency type (i.e. "type@version")
+     * @param _preferredCDN Preferred CDN for dependency type.
+     * @param _preferredRepository Preferred repository for dependency type.
+     */
+    function addDependencyType(
+        bytes32 _dependencyTypeId,
+        string memory _preferredCDN,
+        string memory _preferredRepository,
+        string memory _projectWebsite
+    ) external onlyAdminACL(this.addDependencyType.selector) {
+        require(
+            !_dependencyTypes.contains(_dependencyTypeId),
+            "Dependency type already exists"
+        );
+        require(
+            _dependencyTypeId.containsExactCharacterQty(
+                AT_CHARACTER_CODE,
+                uint8(1)
+            ),
+            "must contain exactly one @"
+        );
+
+        _dependencyTypes.add(_dependencyTypeId);
+        dependencyTypeInfo[_dependencyTypeId].preferredCDN = _preferredCDN;
+        dependencyTypeInfo[_dependencyTypeId]
+            .preferredRepository = _preferredRepository;
+        dependencyTypeInfo[_dependencyTypeId].projectWebsite = _projectWebsite;
+
+        emit DependencyTypeAdded(
+            _dependencyTypeId,
+            _preferredCDN,
+            _preferredRepository,
+            _projectWebsite
+        );
+    }
+
+    /**
+     * @notice Removes a new dependency type.
+     * @param _dependencyTypeId Name of dependency type (i.e. "type@version")
+     */
+    function removeDependencyType(bytes32 _dependencyTypeId)
+        external
+        onlyAdminACL(this.removeDependencyType.selector)
+    {
+        require(
+            _dependencyTypes.contains(_dependencyTypeId),
+            "Dependency type does not exist"
+        );
+
+        _dependencyTypes.remove(_dependencyTypeId);
+        delete dependencyTypeInfo[_dependencyTypeId];
+
+        emit DependencyTypeRemoved(_dependencyTypeId);
+    }
+
+    /**
+     * @notice Returns a list of registered depenency types.
+     * @return List of registered depenency types.
+     */
+    function getRegisteredDependencyTypes()
```

Comment:
I think we should probably make this a "get at index" type of method, to avoid issues with string becoming too big to return.